protocol/shannon/endpoint.go

```diff
@@ -31,49 +119,43 @@ type endpoint struct {
 	session sessiontypes.Session
 }
 
-// TODO_TECHDEBT(@adshmh): Refactor to cleanly separate the "fallback" logic from the endpoint.
```

Comment:
Please keep this TODO and its details: the changes in this PR help, but we need to revisit this.